src/sys/socket/mod.rs

```diff
@@ -983,6 +1015,55 @@ impl ControlMessageOwned {
                 let content_type = unsafe { ptr::read_unaligned(p as *const u8) };
                 ControlMessageOwned::TlsGetRecordType(content_type.into())
             },
+            #[cfg(any(
+                apple_targets,
+                target_os = "freebsd",
+            ))]
+            #[cfg(feature = "net")]
+            (libc::IPPROTO_IP, libc::IP_RECVTOS) => {
+                let tos = unsafe { ptr::read_unaligned(p as *const libc::c_int) };
```

Comment:
I realized that all these pointer type castings are using the `as` keyword, we should replace them with `.cast()` in a separate PR

Reply:
Should I do this in this PR for the additions I made, or leave as is?

Reply:
Leaving them as-is should be ok, we will clear them in another PR